sled-agent/src/services.rs

```diff
@@ -3901,33 +4684,45 @@ impl ServiceManager {
                     );
                 }
 
-                if let Some(info) = self.inner.sled_info.get() {
-                    zone.add_default_route(info.underlay_address).map_err(
-                        |err| Error::ZoneCommand {
-                            intent: "Adding Route".to_string(),
-                            err,
-                        },
-                    )?;
-                }
+                // TODO: This is probably not necessary as it's added byt the zone_network_setup service
+                //    if let Some(info) = self.inner.sled_info.get() {
+                //        zone.add_default_route(info.underlay_address).map_err(
+                //            |err| Error::ZoneCommand {
+                //                intent: "Adding Route".to_string(),
+                //                err,
+                //            },
+                //        )?;
+                //    }
 
                 for service in &request.services {
                     let smfh = SmfHelper::new(&zone, service);
 
                     match service {
                         SwitchService::ManagementGatewayService => {
+                            // TODO: Make sure all services use delpropvalue_default_instance
+                            // and addpropvalue_default_instance instead of delpropvalue
+                            // and addpropvalue. Verify property values are correct
+
                             // Remove any existing `config/address` values
                             // without deleting the property itself.
-                            smfh.delpropvalue("config/address", "*")?;
+                            smfh.delpropvalue_default_instance(
+                                "config/address",
+                                "*",
+                            )?;
 
                             // Restore the localhost address that we always add
                             // when setting up MGS.
-                            smfh.addpropvalue(
+                            smfh.addpropvalue_default_instance(
```

Comment:
Just adding a note here about why I'm doing this. Previously, all services added property values to the service FMRI. Now with the new `PropertyBuilder` flow, we are adding them to the default service _instance_ FMRI. This means that if I didn't change these values to be set directly on the default instance itself, we had a mess of some values set on the instance and some values set on the service directly